Remotion/ObjectBinding/Web.Development.WebTesting/ControlObjects/BocControlObject.cs

```diff
@@ -87,21 +86,20 @@ protected IReadOnlyList<string> GetValidationErrors ([NotNull] ElementScope scop
 
       try
       {
-        var validationErrorIndex = int.Parse (scope[DiagnosticMetadataAttributes.ValidationErrorIDIndex]);
+        var validationErrorIndex = int.Parse (scope.GetAttributeOrThrow (DiagnosticMetadataAttributes.ValidationErrorIDIndex));
+        var validationErrorsScope = Scope.FindId (validationErrorIDs[validationErrorIndex]);
 
-        validationErrorsScope = Scope.FindId (validationErrorIDs[validationErrorIndex]);
+        // re-motion renders the delimiter as <br />, but the browser (and Selenium) show it as <br>
+        const string validationErrorDelimiter = "<br>";

```

Comment:
Why move this inside the try Block? Shouldnt the exception only happen in the Scope.FindId (validationError....) thingy?

Reply:
`FindId (...)` does not synchronize the scope, this happens when synchronizing the scope through calling `.Now()` while accessing `InnerHtml`.

`Coypu/ElementScope.cs`:
```csharp
public string InnerHTML
{
    get
    {
        return Try(() => Now()
                       .InnerHTML);
    }
}
```

Reply:
Oh, that sneaky InnerHTML. I just think it would be cool if only the methods which could throw an MissingHTMLException are inside the Try Scope. But maybe thats too pedantic? I just think it could help understanding, especially as its not clear from the outside which API could throw the exception. Maybe put the the validationErrorIndex declaration in the above try scope and the rest outside? Or does this mess with the flow too much?

Reply:
@FlorianDecker The line

```csharp
var validationErrorIndex = int.Parse (scope.GetAttributeOrThrow (DiagnosticMetadataAttributes.ValidationErrorIDIndex));
```

actually does not belong here. The exception thrown when the `DiagnosticMetadataAttributes.ValidationErrorIDIndex` is missing should not be re-thrown using the validation field exception message, but I might be wrong.

Reply:
Fixed in https://github.com/re-motion/Trunk-SVN2Git/pull/11/commits/b76b9509105123edbd2dc480e96563a9d55edff3.